pkg/webhook/resources/persistentvolumeclaim/validator.go

```diff
@@ -131,90 +130,8 @@ func (v *pvcValidator) Delete(request *types.Request, oldObj runtime.Object) err
 	return nil
 }
 
-func (v *pvcValidator) isOnlineExpandNeeded(pvc *corev1.PersistentVolumeClaim) (bool, error) {
-	vms, err := v.vmCache.GetByIndex(indexeresutil.VMByPVCIndex, ref.Construct(pvc.Namespace, pvc.Name))
-	if err != nil {
-		return false, werror.NewInternalError(fmt.Sprintf("failed to get VMs by index: %s, PVC: %s/%s, err: %s", indexeresutil.VMByPVCIndex, pvc.Namespace, pvc.Name, err))
-	}
-	for _, vm := range vms {
-		if vm.Status.PrintableStatus != kubevirtv1.VirtualMachineStatusStopped {
-			return true, nil
-		}
-	}
-	return false, nil
-}
-
-func (v *pvcValidator) isHotpluggedFilesystemPVC(pvc *corev1.PersistentVolumeClaim) (bool, error) {
-	// Check if the PVC is in Filesystem mode
-	if pvc.Spec.VolumeMode == nil || *pvc.Spec.VolumeMode != corev1.PersistentVolumeFilesystem {
-		return false, nil
-	}
-
-	// Check if the PVC is hotplugged to any VM
-	vms, err := v.vmCache.GetByIndex(indexeresutil.VMByHotplugPVCIndex, ref.Construct(pvc.Namespace, pvc.Name))
-	if err != nil {
-		return false, werror.NewInternalError(err.Error())
-	}
-
-	return len(vms) > 0, nil
-}
-
-func isKubevirtExpandEnabled(kubevirt *kubevirtv1.KubeVirt) bool {
-	featureGates := kubevirt.Spec.Configuration.DeveloperConfiguration.FeatureGates
-	for _, f := range featureGates {
-		if f == kvirtfeatures.ExpandDisksGate {
-			return true
-		}
-	}
-
-	return false
-}
-
 func (v *pvcValidator) checkExpand(pvc *corev1.PersistentVolumeClaim) error {
-	// we will have an early return if there is no related VMs or all related VMs are stopped
-	if onlineExpand, err := v.isOnlineExpandNeeded(pvc); err != nil || !onlineExpand {
-		return err
-	}
-
-	kubevirt, err := v.kubevirtCache.Get(util.HarvesterSystemNamespaceName, util.KubeVirtObjectName)
-	if err != nil {
-		return err
-	}
-	if !isKubevirtExpandEnabled(kubevirt) {
-		return werror.NewInvalidError(util.PVCExpandErrorPrefix+": kubevirt ExpandDisks not included in featureGate", "")
-	}
-
-	hotpluggedFSPVC, err := v.isHotpluggedFilesystemPVC(pvc)
-	if err != nil {
-		return err
-	}
-	if hotpluggedFSPVC {
-		return werror.NewInvalidError(
-			fmt.Sprintf(
-				util.PVCExpandErrorPrefix+": Expansion of hotplugged PVC '%s/%s' in filesystem mode is not supported",
-				pvc.Namespace,
-				pvc.Name,
-			),
-			"",
-		)
-	}
-
-	expandable, err := webhookutil.CheckOnlineExpand(pvc, v.engineCache, v.scCache, v.settingCache)
-	if err != nil {
-		return err
-	}
-	if !expandable {
-		return werror.NewInvalidError(
-			fmt.Sprintf(
-				util.PVCExpandErrorPrefix+": pvc %s/%s is not online expandable with its provider",
-				pvc.Namespace,
-				pvc.Name,
-			),
-			"",
-		)
-	}
-
-	return nil
+	return webhookutil.CheckExpand(pvc, v.vmCache, v.kubevirtCache, v.engineCache, v.scCache, v.settingCache)
```

Comment:
Do we want to use this function directly? It looks like `pvcValidator#checkExpand` only has one line.